### ISSUE TITLE
Backward stage for FakeQuantize 

### ISSUE THREAD

**Issue by jafermarq** [pytorch/pytorch#38600]
## 📚 Documentation

More often than not, integer quantization is performed by "simulating" quantisation during the forward pass and let the gradients flow unchanged during back propagation (i.e. using Straight Through Estimator, STE). 

From the description in [FakeQuantize](https://pytorch.org/docs/stable/_modules/torch/quantization/fake_quantize.html#FakeQuantize) I think it's fair to assume that STE is used since nothing is mentioned about what happens during backward pass. However, FakeQuantize does not use STE. (evidence [here](https://github.com/pytorch/pytorch/blob/fe44741dbac80c73139306f46da8b719d44444d3/aten/src/ATen/native/quantized/cuda/fake_quantize_core.cu#L67) showing the CUDA implementation). The backward pass does gradient clipping. 

I think this should be mentioned (maybe as a "Note"?) in the DOCs since these two approaches for backward prop can lead to very different results in some cases. 

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a

**Comment by vkuzo** [comment]
hi @jafermarq , FakeQuantize does use STE in the backward pass.  The CUDA code you mentioned is zeroing out the gradient only if the forward pass clipped the original value to be inside the quantization params (cc @raghuramank100  to confirm in case I'm misreading the code).

https://arxiv.org/pdf/1806.08342.pdf section 2.4 might also be helpful
